inference-engine/src/mkldnn_plugin/ngraph_transformations/convert_matmul_to_fc_or_gemm.cpp

```diff
@@ -170,7 +175,33 @@ MKLDNNPlugin::ConvertMatMulToGemm::ConvertMatMulToGemm() {
         auto fc_input_a = input_a, fc_input_b = input_b;
         ngraph::NodeVector new_ops;
 
+        if (shape_a.size() == 1) {
+            // If the first input is 1D tensor, it is unsqueezed to 2D tensor (row vector)
+            // by adding axes with size 1 at ROW_INDEX_DIM, to the left of the shape.
+            // For example {S} will be reshaped to {1, S}.
+            fc_input_a = std::make_shared<ngraph::opset1::Unsqueeze>(fc_input_a,
+                ngraph::opset1::Constant::create(ngraph::element::i64, ngraph::Shape{1}, {0}));
+            shape_a = fc_input_a.get_shape();
+            new_ops.push_back(fc_input_a.get_node_shared_ptr());
+            // For 1D inputs transpose flag is expected to always act like `false`
+            matmul->set_transpose_a(false);
+        }
+        if (shape_b.size() == 1) {
+            // If the second input is 1D tensor, it is unsqueezed to 2D tensor (column vector)
+            // by adding axes with size 1 at COL_INDEX_DIM, to the right of the shape.
+            // For example {S} will be reshaped to {S, 1}.
+            fc_input_b = std::make_shared<ngraph::opset1::Unsqueeze>(fc_input_b,
+                ngraph::opset1::Constant::create(ngraph::element::i64, ngraph::Shape{1}, {1}));
+            shape_b = fc_input_b.get_shape();
+            new_ops.push_back(fc_input_b.get_node_shared_ptr());
+            // For 1D inputs transpose flag is expected to always act like `false`
+            matmul->set_transpose_b(false);
+        }
+
         // WA for IE that Gemm must have inputs with the same length.
```

Comment:
Seems, Gemm node was removed. Why do we need this WA?

Reply:
we just renamed Gemm -> MatMul but we still need this